test/jdk/java/net/httpclient/FilePublisher/FilePublisherTest.java

```diff
@@ -156,6 +159,28 @@ public void testZipFs(String uriString,
         send(uriString, path, expectedMsg, sameClient);
     }
 
+    @Test
+    public void testFileNotFound() throws Exception {
+        out.printf("\n\n--- testFileNotFound(): starting\n");
+        var zipPath = Path.of("fileNotFound.zip");
+        try (FileSystem fs = newZipFs(zipPath)) {
+            Path fileInZip = zipFsFile(fs);
+            Files.deleteIfExists(fileInZip);
```

Comment:
Now that I look at this, perhaps we should just change these 2 lines:
```java
Path fileInZip = zipFsFile(fs);
Files.deleteIfExists(fileInZip);
```
 to:

```java
Path fileInZip = fs.getPath("non-existent.txt");
```
The call to `zipFsFile(...)` creates a ZIP entry within the ZIP and we then delete it immediately here with the deleteIfExists(...) call. At least for this new test, this creation followed by the deletion isn't necessary.

Reply:
Simplified as suggested in 1eadf2dc4b8.